### PR TITLE
tests: fix random failures

### PR DESCRIPTION
**Pull request by jma**
* Fixes random errors due to the circulation main component.
* Enables tests for the menu component.
* Creates a new utils file for default mock services such as
RecordService.
* Changes order of operations in the menu component to pass the tests.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- To fix random travis tests failures.

## How to test?

- Run `ng test` several times. No error should happens.
- Open the debug console in the test browser no errors should be there.

## Code review check list

- [x] Commit message template compliance.
- [x] Commit message without typos.
- [x] File names.
- [x] Functions names.
- [ ] Functions docstrings.
- [x] Unnecessary commited files?
- [x] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.spec.ts [hunk 1]
```diff
@@ -18,62 +18,68 @@
...
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
 import { RouterTestingModule } from '@angular/router/testing';
-import { HttpClientModule } from '@angular/common/http';
-import { ToastrModule } from 'ngx-toastr';
-import { RecordModule, RecordService } from '@rero/ng-core';
+import { RecordService, RecordUiService } from '@rero/ng-core';
 import { of } from 'rxjs';
-import { convertToParamMap, ActivatedRoute } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
+import { AppModule } from '../../../app.module';
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.spec.ts:26]: Just sort imports

File: projects/admin/src/app/circulation/patron/main/main.component.spec.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ import { TranslateModule } from '@ngx-translate/core';
 import { CardComponent } from '../card/card.component';
 import { MainComponent } from './main.component';
 import { ToastrModule } from 'ngx-toastr';
+import { RecordService } from '@rero/ng-core';
+import { recordTestingService } from 'projects/admin/tests/utils';
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/patron/main/main.component.spec.ts:26]: Sort imports

File: projects/admin/src/app/menu/menu.component.ts [hunk 3]
```diff
@@ -112,6 +108,21 @@ export class MenuComponent implements OnInit {
...
+    }, {
+      type: 'persons',
+      field: 'autocomplete_name',
+      getSuggestions: (query, persons) => this.getPersonsSuggestions(query, persons)
+    }];
+
+    this._recordService.getRecord('organisations', currentUser.library.organisation.pid)
+    .subscribe(organisation => {
+      this.userMenu.entries[0].entries[0].href = `/${organisation.metadata.code}/`;
+    });
```
Comment (sebastiendeleze) [inline projects/admin/src/app/menu/menu.component.ts:124]: It's not mentioned in the commit message.

File: projects/admin/src/app/circulation/patron/main/main.component.spec.ts [hunk 4]
```diff
@@ -36,9 +38,12 @@ describe('MainComponent', () => {
...
         ToastrModule.forRoot()
       ],
-      declarations: [ MainComponent, CardComponent ]
+      declarations: [MainComponent, CardComponent],
+      providers: [
+        { provide: RecordService, useValue: recordTestingService }
+      ]
     })
-    .compileComponents();
+      .compileComponents();
```
Comment (zannkukai) [inline projects/admin/src/app/circulation/patron/main/main.component.spec.ts:46]: strange indentation ?

File: projects/admin/src/app/menu/menu.component.spec.ts [hunk 5]
```diff
@@ -18,27 +18,30 @@
...
-import { CollapseModule, BsDatepickerModule, TypeaheadModule } from 'ngx-bootstrap';
-import { RecordModule, RecordService } from '@rero/ng-core';
+import { RecordService } from '@rero/ng-core';
 import { TranslateModule } from '@ngx-translate/core';
-import { HttpClientModule } from '@angular/common/http';
-import { BrowserModule } from '@angular/platform-browser';
-import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { UserService } from '../service/user.service';
 import { FormsModule } from '@angular/forms';
 import { RouterTestingModule } from '@angular/router/testing';
```
Comment (zannkukai) [inline projects/admin/src/app/menu/menu.component.spec.ts:25]: bad sorting imports :-)

File: projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.spec.ts [hunk 6]
```diff
@@ -18,62 +18,68 @@
...
         TranslateService,
-        { provide: RecordService, useValue: recordService },
-        { provide: ActivatedRoute, useValue: route }
+        { provide: RecordService, useValue: recordTestingService },
+        { provide: RecordUiService, useValue: recordTestingUiService },
+        { provide: ActivatedRoute, useValue: testingRoute }
       ]
     })
-    .compileComponents();
+      .compileComponents();
```
Comment (zannkukai) [inline projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.spec.ts:82]: strange indentation ?

**Pull request merged**